### PR TITLE
[PB-2385]:feat/Refactor url navigation in workspaces

### PR DESCRIPTION
**Pull request by Jona-Internxt**

### CODE REVIEW

**Review by rafijv — APPROVED** [review]

File: src/app/core/components/Sidenav/Sidenav.tsx [hunk 1]
```diff
@@ -97,26 +97,28 @@ const Sidenav = ({
...
+      to: workspaceUuid ? `/?workspaceid=${workspaceUuid}` : '/',
       isActive: isActiveButton('/') || isActiveButton('/file/:uuid') || isActiveButton('/folder/:uuid'),
       label: translate('sideNav.drive'),
       icon: FolderSimple,
       iconDataCy: 'sideNavDriveIcon',
       isVisible: true,
     },
     {
-      to: '/backups',
+      to: workspaceUuid ? `/backups/?workspaceid=${workspaceUuid}` : '/backups',
```
Comment (CandelR) [inline src/app/core/components/Sidenav/Sidenav.tsx]: backups and recents views are not in b2b. 
Maybe could create a function that generates the workspace url for a given path

File: src/app/core/services/navigation.service.ts [hunk 2]
```diff
@@ -81,6 +84,14 @@ const navigationService = {
       errorService.reportError(error);
     }
   },
+  setWorkspaceFromParams(workspaceThunks, dispatch: AppDispatch): void {
+    const params = new URLSearchParams(window.location.search);
+    const currentWorkspaceUuid = params.getAll('workspaceid');
+    currentWorkspaceUuid.length === 1 &&
```
Comment (CandelR) [inline src/app/core/services/navigation.service.ts]: maybe could extract the conditional `currentWorkspaceUuid.length === 1` to a const in order to give a proper name to understand the reason

Comment (Jona-Internxt) [reply src/app/core/services/navigation.service.ts]: I think it's clearer with this change

File: src/app/core/components/Sidenav/Sidenav.tsx [hunk 3]
```diff
@@ -84,6 +84,9 @@ const SideNavItems = ({ sideNavItems }: { sideNavItems: SideNavItemsProps[] }) =
     ))}
   </>
 );
+const getItemNavigatoinPath = (path: string, workspaceUuid?: string) => {
```
Comment (CandelR) [inline src/app/core/components/Sidenav/Sidenav.tsx]: typo: `getItemNavigatoinPath` -> `getItemNavigationPath`

**Review by CandelR — APPROVED** [review]

**Pull request merged**